src/control_connection.cpp

```diff
@@ -395,12 +395,10 @@ void ControlConnection::handle_refresh_node(RefreshNodeCallback* callback) {
   const Row* row = NULL;
   ResultIterator rows(callback->result().get());
 
-  while (rows.next() && !found_host) {
+  while (!found_host && rows.next()) {
     row = rows.row();
     if (callback->is_all_peers) {
```

Comment:
would `callback->is_all_peers` not be a constant for the duration fo the while loop? Maybe we could have something like
`
bool found_host = callback->is_all_peers && callback->result().row_count() > 0;
while (!found_host && rows.next()) {
  row = rows.row();
  found_host = settings_.address_factory->is_peer(row, connection_->host(), callback->address);
}
`

Reply:
Nice suggestion.